front/src/features/books/booksSlice.js

```diff
@@ -0,0 +1,79 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+import api from "api";
+
+export const booksSlice = createSlice({
+  name: "books",
+  initialState: {
+    data: [],
+    basket: [],
+    basketValue: 0,
+    dataStatus: "iddle",
+  },
+  reducers: {
+    fetchBooksDataStarted: (state) => {
+      state.dataStatus = "inProgress";
+    },
+    fetchBooksDataDataSucceeded: (state) => {
+      state.dataStatus = "succeeded";
+    },
+    fetchBooksDataDataFailed: (state) => {
+      state.dataStatus = "failed";
+    },
+    setData: (state, action) => {
+      state.data = action.payload.data;
+    },
+    addItemToBasket: (state, action) => {
+      const IsInBasket = state.basket.find(
+        (item) => item.id === action.payload.id
+      );
+
+      if (IsInBasket) {
+        state.basket = [
```

Comment:
mozesz to zrobic krocej uzywajac ternary + staraj sie zawsze opisywac elementy w mapie, tj. basket to zbior elementow basketItem, ew. product. 
books to zbiór elementów book, etc.

state.basket = state.basket.map((basketItem) => 
element.id === action.payload.id ? 
{
 ...basketItem,
value: element.value + 1,
}
: basketItem)

Reply:
Mega zajebisty comment 👍

Reply:
Done.